components/NavPage.qml

```diff
@@ -0,0 +1,201 @@
+import QtQuick 2.0
+import QtLocation 5.0
+import Ubuntu.Components 1.1
+import Ubuntu.Components.Popups 1.0
+import Ubuntu.Components.ListItems 1.0 as ListItem
+import QtQuick.XmlListModel 2.0
+import QtQuick.LocalStorage 2.0
+import "../models" as Models
+import "./Helpers.js" as Helpers
+
+Page {
+    id: navPage
+    title: i18n.tr("Navigate")
+    visible: false
+
+    property string navState: ""
+    property real fromlon: 0.0
+    property real fromlat: 0.0
+    property real tolon: 0.0
+    property real tolat: 0.0
+
+    Models.SearchPlaceModel {
+        id: searchModel
+        onStatusChanged: {
+            if (status === XmlListModel.Error) {
+                httpFailedSearch.show();
+            }
+            if (status === XmlListModel.Ready && count === 0) {
+                var place = "your search";
+                if (navState === "startpoint") {
+                    place = startPoint.text;
+                } else if (navState === "endpoint") {
+                    place = endPoint.text;
+                }
+
+                noResult.text = i18n.tr("Sorry, no result for ") + place;
+                noResult.visible = true;
+            } else {
+                noResult.visible = false;
+            }
+        }
+    }
+
+    SplashComponent {
+        id: httpFailedSearch
+        objectName: "HTTPFailedSearch"
+    }
+
+    TextField {
+        id: startPoint
+        width: parent ? parent.width - units.gu(2) : undefined
+        hasClearButton: true
+        focus: true
+        placeholderText: i18n.tr("Enter a place name to start from")
+        primaryItem: Image {
+            height: parent.height*0.5
+            width: parent.height*0.5
+            anchors.verticalCenter: parent.verticalCenter
+            anchors.verticalCenterOffset: -units.gu(0.2)
+            source: Helpers.getIcon("search")
+        }
+        onAccepted: {
```

Comment:
That's a good idea. That'd be done from the onClicked handler in the ListView, I think. I wondered about disabling the text area when the Enter is pressed, but then it's not possible to change your mind and modify the endpoints. Just changing the text would work better.